### ISSUE TITLE
MPD 0.21.11 soxr resampling hangs on seek

### ISSUE THREAD

**Issue by ClaudioRototype** [MusicPlayerDaemon/MPD#612]
## Bug report
(I apologize for previous sending without proper template, I had clicked on Submit an Issue in MPD website instead of using GitHub interface).
### Describe the bug
When SOXR resampler is used, SEEK command hangs MPD.
I have the following in the configuration file:

----------------------------------
```
audio_output {
	type		"alsa"
	name		"My ALSA Device"
	device		"hw:0,0"
	mixer_type      "none"      	# optional
	auto_resample	"no"
	dop		"no"
	allowed_formats "96000:*:* 88200:*:*"
}
resampler {
	plugin "soxr"
	quality "very high"
}
```
----------------------------------
I start playing a CD rip track (44.1 16 bit FLAC) that is resampled to 96000 by soxr. Then I seek 1 minute forward:
```
root@voyage:/etc# mpc play
The Royal Philharmonic Orchestra - In The Air Tonight
[playing] #1/10   0:00/5:06 (0%)
volume: n/a   repeat: off   random: off   single: off   consume: off
root@voyage:/etc# mpc seek +00:01:00
The Royal Philharmonic Orchestra - In The Air Tonight
[playing] #1/10   1:19/5:06 (25%)
volume: n/a   repeat: off   random: off   single: off   consume: off
root@voyage:/etc# mpc
error: Connection refused
```
Music stops and MPD seems to be hanging (see: reply to mpc is "connection refused").

## Expected Behavior
Seek should work with resampling too.

## Actual Behavior
Soxr resampler causes hanging

## Version
```
root@voyage:~# mpd --version
Music Player Daemon 0.21.11 (0.21.11)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2018 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Database plugins:
 simple proxy upnp
Storage plugins:
 local smbclient udisks nfs curl
Neighbor plugins:
 smbclient upnp udisks
Decoders plugins:
 [mad] mp3 mp2
 [mpg123] mp3
 [vorbis] ogg oga
 [oggflac] ogg oga
 [flac] flac
 [opus] opus ogg oga
 [sndfile] wav aiff aif au snd paf iff svx sf voc w64 pvf xi htk caf sd2
 [audiofile] wav au aiff aif
 [dsdiff] dff
 [dsf] dsf
 [hybrid_dsd] m4a
 [faad] aac
 [mpcdec] mpc
 [wavpack] wv
 [modplug] 669 amf ams dbm dfm dsm far it med mdl mod mtm mt2 okt s3m stm ult umx xm
 [mikmod] amf dsm far gdm imf it med mod mtm s3m stm stx ult uni xm
 [sidplay] sid mus str prg P00
 [wildmidi] mid
 [fluidsynth] mid
 [adplug] amd d00 hsc laa rad raw sa2
 [ffmpeg] 16sv 3g2 3gp 4xm 8svx aa3 aac ac3 adx afc aif aifc aiff al alaw amr anim apc ape asf atrac au aud avi avm2 avs bap bfi c93 cak cin cmv cpk daud dct divx dts dv dvd dxa eac3 film flac flc fli fll flx flv g726 gsm gxf iss m1v m2v m2t m2ts m4a m4b m4v mad mj2 mjpeg mjpg mka mkv mlp mm mmf mov mp+ mp1 mp2 mp3 mp4 mpc mpeg mpg mpga mpp mpu mve mvi mxf nc nsv nut nuv oga ogm ogv ogx oma ogg omg opus psp pva qcp qt r3d ra ram rl2 rm rmvb roq rpl rvc shn smk snd sol son spx str swf tak tgi tgq tgv thp ts tsp tta xa xvid uv uv2 vb vid vob voc vp6 vmd wav webm wma wmv wsaud wsvga wv wve
 [gme] ay gbs gym hes kss nsf nsfe sap spc vgm vgz
 [pcm]
Filters:
 libsamplerate soxr
Tag plugins:
 id3tag
Output plugins:
 shout null fifo pipe alsa ao oss openal pulse jack httpd recorder
Encoder plugins:
 null vorbis opus lame wave flac
Archive plugins:
 [bz2] bz2
 [zzip] zip
 [iso] iso
Input plugins:
 file archive alsa tidal qobuz curl ffmpeg smbclient nfs mms cdio_paranoia
Playlist plugins:
 extm3u m3u pls xspf asx rss soundcloud flac cue embcue
Protocols:
 file:// alsa:// tidal:// qobuz:// http:// https:// gopher:// rtp:// rtsp:// rtmp:// rtmpt:// rtmps:// smb:// nfs:// mms:// mmsh:// mmst:// mmsu:// cdda://
Other features:
 avahi dbus udisks epoll icu inotify ipv6 systemd tcp un
```

## Log
```
Aug 02 21:35 : client: [1] process command "play"
Aug 02 21:35 : playlist: play 0:"usbmount/usb0/C_MusicNQ/Royal Philharmonic orchestra - Hits of Phil Collins/01. The Royal Philharmonic Orchestra, Louis Clark - In The Air Tonight.flac"
Aug 02 21:35 : decoder_thread: probing plugin flac
Aug 02 21:35 : client: [1] command returned 0
Aug 02 21:35 : playlist: queue song 1:"usbmount/usb0/C_MusicNQ/Royal Philharmonic orchestra - Hits of Phil Collins/02. The Royal Philharmonic Orchestra, Louis Clark - Groovy Kind Of Love.flac"
Aug 02 21:35 : decoder: audio_format=44100:16:2, seekable=true
Aug 02 21:35 : client: [1] process command list
Aug 02 21:35 : client: process command "status"
Aug 02 21:35 : alsa_output: opened hw:0,0 type=HW
Aug 02 21:35 : alsa_output: buffer: size=192..174762 time=2000..1820438
Aug 02 21:35 : alsa_output: period: size=96..87381 time=1000..910219
Aug 02 21:35 : alsa_output: default period_time = buffer_time/4 = 500000/4 = 125000
Aug 02 21:35 : alsa_output: format=S24_3LE (Signed 24 bit Little Endian in 3bytes)
Aug 02 21:35 : alsa_output: buffer_size=48000 period_size=12000
Aug 02 21:35 : output: opened "My ALSA Device" (alsa) audio_format=96000:24:2
Aug 02 21:35 : soxr: soxr engine 'double-precision'
Aug 02 21:35 : soxr: samplerate conversion ratio to 2.18
Aug 02 21:35 : output: converting in=44100:16:2 -> f=44100:16:2 -> out=96000:24:2
Aug 02 21:35 : client: command returned 0
Aug 02 21:35 : client: process command "currentsong"
Aug 02 21:35 : client: command returned 0
Aug 02 21:35 : client: [1] process command list returned 0
Aug 02 21:35 : client: [1] closed
Aug 02 21:35 : state_file: Saving state file /var/lib/mpd/state
Aug 02 21:35 : client: [2] opened from 127.0.0.1:55761
Aug 02 21:35 : client: [2] process command "status"
Aug 02 21:35 : client: [2] command returned 0
Aug 02 21:35 : client: [2] process command "seekid "1" "79""
Aug 02 21:35 : playlist: queue song 1:"usbmount/usb0/C_MusicNQ/Royal Philharmonic orchestra - Hits of Phil Collins/02. The Royal Philharmonic Orchestra, Louis Clark - Groovy Kind Of Love.flac"
Aug 02 21:35 : client: [2] command returned 0
Aug 02 21:35 : client: [2] process command list
Aug 02 21:35 : client: process command "status"
Aug 02 21:35 : client: command returned 0
Aug 02 21:35 : client: process command "currentsong"
Aug 02 21:35 : client: command returned 0
Aug 02 21:35 : client: [2] process command list returned 0
Aug 02 21:35 : client: [2] closed
```

**Comment by MaxKellermann** [comment]
Log is still missing!

**Comment by ClaudioRototype** [comment]
You are too fast :-). I was still editing....

**Comment by ClaudioRototype** [comment]
Now it should be complete; I had to replicate it again because I had not realized I had to set the log mode to verbose to have something useful in there. Anyway I cannot see a clear indication of the hang in the log.
Additional information: mpd does not seem to be consuming much cpu (not showing in top).

**Comment by MaxKellermann** [comment]
I still cannot reproduce this.
But if you say you get "Connection refused", then MPD doesn't "hang" - this implies that the listener socket is gone, which can be because MPD has crashed.
Does the MPD process still run? Did it crash? What did it print (to console/journal) before it died?
If it crashed, I need a backtrace.

**Comment by ClaudioRototype** [comment]
can you keep this on hold for two weeks? I think you are right but I have no access to the system to provide backtrace at the moment and I need more time to investigate. 
The console interaction that I reported is complete: the reply to local mpc seek +1 min command is coherent (current time plus 1 minute, name of song) as if command was accepted and executed. But music stops playing and mpd is down.
I wrote Hanging because mpd restart says "stopped correctly" and I thought that as an indication that it was still running but it doesn't show in top (sorry I have not checked, and will check, ps) and I think you are right that it is a crash instead. Since it only happens when resampling with soxr (not happening playing an allowed format) it should be a libsoxr crash.
But I thought it could be a general issue. If you cannot replicate, it might be just my installation (an outdated library?). I want to check better and will come back with more information because, if it is my issue, you should not waste time. 

Thanks and all the best
Claudio

Il 3 agosto 2019 12:43:55 CEST, Max Kellermann <notifications@github.com> ha scritto:

**Comment by MaxKellermann** [comment]
Let's get this bug report out of the way until there is enough information.

**Comment by ClaudioRototype** [comment]
Dear Mr. Kellermann,
I double checked and it is a crash as per your assumption (mpd is no longer in the running processes after the seek command).
It is a crash in libsoxr (it only happens when resampling with libsoxr is in place (not happening with libsample rate).
The last lines of syslog are:
---
Aug 25 16:39:01 localhost CRON[3309]: (root) CMD (  [ -x /usr/lib/php5/sessionclean ] && /usr/lib/php5/sessionclean)
Aug 25 16:58:01 localhost vmunix: [13023.867254] usb 1-6: 1:1: cannot get freq at ep 0x1
Aug 25 16:58:17 localhost vmunix: [13040.152959] output:My ALSA [3022]: segfault at 0 ip b5316bc7 sp aacfe8c0 error 4
Aug 25 16:58:17 localhost vmunix: [13040.152976] output:My ALSA [3023]: segfault at 4 ip b5316bc7 sp aa4fdde0 error 4 in libsoxr.so.0.1.0[b5315000+30000]
Aug 25 16:58:17 localhost vmunix: [13040.152991]  in libsoxr.so.0.1.0[b5315000+30000]
---

**Comment by ClaudioRototype** [comment]
Dear Mr. Kellermann,
After another couple of hours of tries I solved the problem that affected my installation. Thanks for pointing the way by suggesting to look for a crash report.
I updated manually libsoxr-dev package and the problem has disappeared.
I don't know if libsoxr-dev should be considered a dependency and should be updated automatically when updating mpd or not, maybe is just my ignorance about Linux installation process. I thought that updating mpd (apt-get install mpd) would automatically update the dependencies.
But if there is a problem (beyond my ignorance) this is with Debian and not with MPD.
Thanks for your patience and help.
I would appreciate your comment anyway (even if that would be a blaming one for having fallen in a trivial trick).

Regards
Claudio
